Reminder Listing Page

bumbledog opened the pull request:
This adds a very simple frontend UI and class functionality to the app.
It's not pretty at all yet (I'm still working on that, I wasn't liking any styles I tried yet) but it's a solid start to work off of.

Commit:
Reminder Listing Page

samschaevitz commented on Reminder/app/src/main/java/com/reminder/app/ReminderList.java:
```diff
@@ -0,0 +1,60 @@
+        addActionListeners();
+    }
+
+    private void initLayout(){
```
nit: space before the opening '{'
http://blog.codinghorror.com/new-programming-jargon/ (see: egyptian brackets) -- they come from K&R (http://en.wikipedia.org/wiki/Indent_style#Variant:_1TBS)

samschaevitz commented on Reminder/app/src/main/java/com/reminder/app/ReminderList.java:
```diff
@@ -0,0 +1,60 @@
+
+
+    //Not entirely sure this is needed for the final , this'll probably get taken car of on another page
+    protected void addReminder(String reminder){
```
nit: space before the opening '{'

samschaevitz commented on Reminder/app/src/main/java/com/reminder/app/ReminderList.java:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    //various action listeners for the widgets
+    protected void addActionListeners(){
```
nit: space before the opening '{'

samschaevitz commented on Reminder/app/src/main/java/com/reminder/app/ReminderList.java:
```diff
@@ -0,0 +1,60 @@
+    private Button addButton;
+    protected ReminderListAdapter reminderAdapter;
+    protected ArrayList<Reminder> reminderArray;
+    @Override
```
nit: add a newline before method declaration (between current lines 19 and 20)

samschaevitz commented on Reminder/app/src/main/java/com/reminder/app/ReminderList.java:
```diff
@@ -0,0 +1,60 @@
+        setContentView(R.layout.default_page);
+        reminderList = (ListView)findViewById(R.id.reminderList);
+        addButton = (Button)findViewById(R.id.newReminderButton);
+        //temporary List of reminders to work with - replaced when we get add functionality and local storage
```
nit: space between '//' and comment.

samschaevitz commented on Reminder/app/src/main/java/com/reminder/app/ReminderList.java:
```diff
@@ -0,0 +1,60 @@
+        //temporary List of reminders to work with - replaced when we get add functionality and local storage
+        Resources object = this.getResources();
+        String[] reminderStrings = object.getStringArray(R.array.reminderList);
+        //fill the ListArray
```
nit: space between '//' and comment.

samschaevitz commented on Reminder/app/src/main/java/com/reminder/app/ReminderList.java:
```diff
@@ -0,0 +1,60 @@
+    }
+
+
+    //Not entirely sure this is needed for the final , this'll probably get taken car of on another page
```
nit: space between '//' and comment.

samschaevitz commented on Reminder/app/src/main/java/com/reminder/app/ReminderList.java:
```diff
@@ -0,0 +1,60 @@
+        reminderAdapter.notifyDataSetChanged();
+    }
+
+    //various action listeners for the widgets
```
nit: space between '//' and comment.

samschaevitz commented on Reminder/app/src/main/java/com/reminder/app/ReminderList.java:
```diff
@@ -0,0 +1,60 @@
+    protected void addActionListeners(){
+        addButton.setOnClickListener(new View.OnClickListener() {
+            public void onClick(View view) {
+                //TODO: Button click moves to add reminder page
```
Let's give the TODO an owner :) e.g. '// TODO(kylie): Button click...'

samschaevitz commented on Reminder/app/src/main/java/com/reminder/app/ReminderListAdapter.java:
```diff
@@ -0,0 +1,47 @@
+/**
+ * Created by Kylie Williamson on 6/15/14.
+ */
+public class ReminderListAdapter extends BaseAdapter{
```
nit: space before the opening '{'

samschaevitz replied:
Last one of these for now, I think.

samschaevitz commented on Reminder/app/src/main/java/com/reminder/app/Reminder.java:
```diff
@@ -0,0 +1,56 @@
+    protected int latitude;
+    protected String reminderText;
+
+    Reminder(String reminder){
```
nit: space before the opening '{'

samschaevitz commented on Reminder/app/src/main/java/com/reminder/app/Reminder.java:
```diff
@@ -0,0 +1,56 @@
+    }
+
+    public int getUrgency() {
+        return urgency;
```
should this be 'this.urgency'?

samschaevitz replied:
update: i am re-reading java docs as I haven't written java in too long. 

let's explicitly agree on a common style for when we're going to use 'this.' to fetch class vars. I don't care too strongly either way :) you can leave this as-is if it suits you.

samschaevitz replied:
I will state (for the record) my preference on this:
Since (in theory) you logically will Get a value after you Set it, it flows a bit better in my head that you get the this.variable value, as you could have Set it to something after the initial construction. 

I'd play with this and make sure that your Getters get the right values after you Set them.

bumbledog replied:
In this case, the value is the same as this.value. All other variables by that name have since gone out of scope. It shouldn't matter for anything except style. The difference is in the setters, where I have a class level and method level of the same name, and it will default to the method level variable without a this.

samschaevitz commented on Reminder/app/src/main/java/com/reminder/app/ReminderView.java:
```diff
@@ -0,0 +1,38 @@
+
+    public ReminderView(Context context, Reminder reminder) {
+        super(context);
+        //defines it by the XML
```
nit: space between '//' and comment.

samschaevitz commented on Reminder/app/src/main/java/com/reminder/app/ReminderView.java:
```diff
@@ -0,0 +1,38 @@
+        //defines it by the XML
+        LayoutInflater inflater = (LayoutInflater)context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+        inflater.inflate(R.layout.reminder, this, true);
+        //find the correct items
```
nit: space between '//' and comment.

samschaevitz commented:
Lookin' good! A couple of stylistic comments on spacing and whatnot. I'm also assuming y'all are testing this for functionality so I'll leave that to you :)

Commit:
Fixed the stylistic errors.

Commit:
Build\intermediates files

Not entirely sure what these do, but they probably need to be committed
since they're in the project

Commit:
Build\intermediates files part 2

commit didn't seem to work last time

bumbledog commented:
Jordan, you shouldn't be seeing any red in the files. I'm not entirely sure whats going wrong there.

bumbledog commented:
Anyways, I've fixed most of the things. I'm committing a second constructor for the reminder class that can set all of the values. We might eventually delete the plain text one, but for now you're right, it just makes our lives easier.

Commit:
Adding second constructor for reminder class.

bumbledog commented:
The one thing I am afraid of, I am unable to commit my build\intermediary files at the moment, and it might be giving Jordan errors.

jomivega commented:
For now, after talking to Sam, we shouldnt worry too much about the red we are seeing in the  xml, iml and .gradle. They should be resolved when we build in our own IDE.

jomivega commented:
"The one thing I am afraid of, I am unable to commit my build\intermediary files at the moment, and it might be giving Jordan errors." Per this comment, have you been able to commit those files? I havent tried running this yet.

bumbledog commented:
I haven't. I figured they might not be necessary (compile created files)

Sent from Yahoo! Mail on Android

jomivega commented:
Right, I will try running this!

samschaevitz commented:
If it builds (and you fix that last space for now), LGTM!

bumbledog commented:
I must have missed that last space! I'll fix that. Jordan, is it working for you?

jomivega commented:
Let me update my local with your fixes and try again. Will do so when I get
home.
On Jun 25, 2014 2:00 PM, "Kylie Williamson" notifications@github.com
wrote:

> I must have missed that last space! I'll fix that. Jordan, is it working
> for you?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/GoogleJump/PinkFairyArmadillos/pull/2#issuecomment-47158017
> .

jomivega commented:
Looks good to me. Go ahead and merge.

Commit:
Adding style change
